tensorlayer/db.py

```diff
@@ -5,11 +5,725 @@
+        """Delete datasets.
+
+        Parameters
+        -----------
+        kwargs : logging information
+            Find items to delete, leave it empty to delete all log.
+        """
+        self._fill_project_info(kwargs)
+        self.db.Dataset.delete_many(kwargs)
+        logging.info("[Database] Delete Dataset SUCCESS")
+
+    ## =========================== LOGGING =============================== ##
+    def train_log(self, **kwargs):
+        """Saves the training log, timestamp will be added automatically.
+
+        Parameters
+        -----------
+        kwargs : logging information
+            Events, such as accuracy, loss, step number and etc.
+
+        Examples
+        ---------
+        >>> db.train_log(accuray=0.33, loss=0.98)
+        """
+        self._fill_project_info(kwargs)
+        kwargs.update({'time': datetime.utcnow()})
+        _result = self.db.TrainLog.insert_one(kwargs)
+        _log = self._print_dict(kwargs)
+        logging.info("[Database] train log: " + _log)
+
+    def valid_log(self, **kwargs):
+        """Saves the validation log, timestamp will be added automatically.
+
+        Parameters
+        -----------
+        kwargs : logging information
+            Events, such as accuracy, loss, step number and etc.
+
+        Examples
+        ---------
+        >>> db.valid_log(accuray=0.33, loss=0.98)
+        """
+        self._fill_project_info(kwargs)
+        kwargs.update({'time': datetime.utcnow()})
+        _result = self.db.ValidLog.insert_one(kwargs)
+        _log = self._print_dict(kwargs)
+        logging.info("[Database] valid log: " + _log)
+
+    def test_log(self, **kwargs):
+        """Saves the testing log, timestamp will be added automatically.
+
+        Parameters
+        -----------
+        kwargs : logging information
+            Events, such as accuracy, loss, step number and etc.
+
+        Examples
+        ---------
+        >>> db.test_log(accuray=0.33, loss=0.98)
+        """
+        self._fill_project_info(kwargs)
+        kwargs.update({'time': datetime.utcnow()})
+        _result = self.db.TestLog.insert_one(kwargs)
+        _log = self._print_dict(kwargs)
+        logging.info("[Database] test log: " + _log)
+
+    def del_train_log(self, **kwargs):
+        """Deletes training log.
+
+        Parameters
+        -----------
+        kwargs : logging information
+            Find items to delete, leave it empty to delete all log.
+
+        Examples
+        ---------
+        - Save training log
+        >>> db.train_log(accuray=0.33)
+        >>> db.train_log(accuray=0.44)
+
+        - Delete logs that match the requirement
+        >>> db.del_train_log(accuray=0.33)
+
+        - Delete all logs
+        >>> db.del_train_log()
+        """
+        self._fill_project_info(kwargs)
+        self.db.TrainLog.delete_many(kwargs)
+        logging.info("[Database] Delete TrainLog SUCCESS")
+
+    def del_valid_log(self, **kwargs):
```

Comment:
instead of have 'delete_train_log', 'dlkete_validation_log', and 'delete_test_log'

can we have a unified "save_event(log)" and 'delete_event(log)' and in the event where you specify the phase of the events: 'training', 'validation', and 'testing'